app/controllers/patients_controller.rb

```diff
@@ -55,6 +55,31 @@ def download
     end
   end
 
+  def fetch_pledge
+    endpoint = ENV.fetch('PLEDGE_GENERATOR_ENDPOINT', 'http://localhost:3001/generate')
+    basic_auth = { username: ENV.fetch('PLEDGE_GENERATOR_USER', 'apiuser'), password: ENV.fetch('PLEDGE_GENERATOR_TOKEN', 'PLEDGETOKEN') }
+    extra_params = params.permit(*current_tenant.pledge_config.remote_pledge_extras.map { |x, y| x.to_sym })
+    payload = {
+      fund: Rails.env.development? ? 'test' : current_tenant.name.downcase,
+      base: {
```

Comment:
the idea here is that some of these every fund is going to need (patient name, who's receiving the $, etc) and some are very fund specific. This is trying to set this up to be as dynamic as possible, and my hope is that when it's situated it won't require code changes in this codebase. I do think it might take some time for what's in `base` to settle though, so please don't think about that too hard while we're reviewing.